library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -61,25 +70,48 @@ public void setup(PortContext context)
     }
   };
 
-  @OutputPortFieldAnnotation(optional = true)
   public transient DefaultOutputPort<INPUT> err = new DefaultOutputPort<INPUT>();
```

Comment:
Certainly, logs can contain stack trace. But a short error message on error port along with error tuple would be useful for users to quickly find out validation errors by connecting to the error port instead of digging into the logs.
What i suggest, the port should at least have provision to send error message.